sky/clouds/aws.py

```diff
@@ -461,6 +461,18 @@ def check_credentials(cls) -> Tuple[bool, Optional[str]]:
 
     @classmethod
     def _current_identity_type(cls) -> Optional[AWSIdentityType]:
+        credentials = aws.session().get_credentials()
+        if credentials is not None and credentials.token is not None:
+            # We have seen the following case: user is using an Okta wrapper
+            # which generates temporary env vars for AWS credentials (including
+            # AWS_SESSION_TOKEN). This technically is using SSO, but the `aws
+            # configure list` output check below would show `env` in the Type
+            # column, rather than `sso` that we expect; hence it would fail to
```

Comment:
If this is the case, should we only apply the check when the column shows `env`, so that we make sure we are checking the same credential as the actual credential used by aws cli?

Reply:
What do you mean by 

> so that we make sure we are checking the same credential as the actual credential used by aws cli

?

Reply:
Oh, I was wondering what would happen if the IAM role is used, but the environment variable `AWS_SESSION_TOKEN` is somehow also specified. In that case, I suppose `aws configure list` will show `iam-role`, but will the `credentials.token` be not None as well?

Reply:
Great catch! I've updated the fix. PTAL.